plugin/src/test/java/testcode/spring/SpringUnvalidatedRedirectController.java

```diff
@@ -0,0 +1,34 @@
+package testcode.spring;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+public class SpringUnvalidatedRedirectController {
+
+    @RequestMapping("/redirect1")
+    public String redirect1(@RequestParam("url") String url) {
```

Comment:
It should detect redirects for String and ModelAndView return values.  Essentially, the detector looks for string concatenation using the "redirect:" prefix in Spring controllers.  This could lead to false positives, but there should be limited false negatives.  I have updated the tests to confirm it will detect ModelAndViews and added LGPL license headers.